topo_processor/metadata/metadata_validators/metadata_validator_stac.py

```diff
@@ -49,6 +50,23 @@ def validate_metadata(self, item: Item) -> None:
             except STACValidationError as e:
                 raise STACValidationError(message=f"Not valid STAC: {e}")
 
+    def validate_metadata_pystac_collection(self, collection: PystacCollection) -> None:
```

Comment:
You should re-use the existing `validate_metadata` method (change the signature and refactor the callers) to avoid code duplication here.